x-pack/plugins/embeddable_enhanced/public/plugin.ts

```diff
@@ -49,16 +53,19 @@ export interface StartDependencies {
 // eslint-disable-next-line @typescript-eslint/no-empty-interface
 export interface SetupContract {}
 
+export interface ReactEmbeddableDynamicActionsApi {
+  dynamicActionsApi: HasDynamicActions;
+  dynamicActionsComparator: StateComparators<DynamicActionsSerializedState>;
+  serializeDynamicActions: () => DynamicActionsSerializedState;
+  startDynamicActions: () => { stopDynamicActions: () => void };
+}
```

Comment:
Originally, we were starting the `DynamicActionManager` in the `initializeDynamicActions` method - however, we weren't **stopping** the manager on unmount which this resulted in drilldowns being duplicated as you navigated around. This cleanup needed to be handled by the embeddable itself (since we have no way of determining where the unmount happens from this plugin), so I knew that the `stop` method would have to be made available in the returned API somehow.

It felt cleaner to me that, rather than still starting the manager in `initializeDynamicActions` and returning **just** the stop method in this API, we tie these two things together - that way, it should be clear that the embeddable is responsible for **both** starting and stopping the manager.